src/VecSim/algorithms/hnsw/hnsw.h

```diff
@@ -935,27 +954,21 @@ idType HNSWIndex<DataType, DistType>::mutuallyConnectNewElement(
     size_t max_M_cur = level ? M : M0;
 
     // Filter the top candidates to the selected neighbors by the algorithm heuristics.
-    getNeighborsByHeuristic2(top_candidates, M);
-    assert(top_candidates.size() <= M &&
-           "Should be not be more than M candidates returned by the heuristic");
-
-    // Hold (distance_from_new_node_id, neighbor_id) pair for every selected neighbor.
-    vecsim_stl::vector<std::pair<DistType, idType>> selected_neighbors(this->allocator);
-    selected_neighbors.reserve(M);
-    while (!top_candidates.empty()) {
-        selected_neighbors.push_back(top_candidates.top());
-        top_candidates.pop();
-    }
-
+    candidatesList<DistType> top_candidates_vec(this->allocator);
```

Comment:
also,vector has a copy ctor
```suggestion
    candidatesList<DistType> top_candidates_list(this->allocator);
```

Reply:
will change the name. Cant use a copy constructor from a priority queue though :). This is one of the places I use the random iterator